From 2122fc9eccefd381db7c830045c4fa65d2bffa63 Mon Sep 17 00:00:00 2001
From: Simeon Bird <spb@ias.edu>
Date: Sat, 8 Dec 2012 23:31:23 -0500
Subject: [PATCH 10/10] Avoid crash on destruction of ClientModel if nepomuk
 was disabled during the lifetime of the ClientModel.

Crash is also fixed with 690e59e13a applied.

REVIEW: 107639
BUG: 311222
---
 client/clientmodel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/clientmodel.cpp b/client/clientmodel.cpp
index bdcbed3..4ac406b 100644
--- a/client/clientmodel.cpp
+++ b/client/clientmodel.cpp
@@ -50,7 +50,7 @@ Soprano::Client::ClientModel::~ClientModel()
     // closed by the server anyway.
     //
     QMutexLocker locker( &m_openIteratorsMutex );
-    if ( m_client->isConnected() ) {
+    if ( m_client && m_client->isConnected() ) {
         for ( int i = 0; i < m_openIterators.count(); ++i ) {
             m_client->iteratorClose( m_openIterators[i] );
         }
-- 
1.8.0.2

